### DISCUSSION TITLE
Query Overload?: useSWR as an alternative to Subscriptions

### DISCUSSION THREAD

**Discussion by taylor-lindores-reeves** [vercel/next.js#27240] [Help]
TL;DR

I have a client project (an auction site) which I have had some serious trouble with that has involved lawyers. Due to the nature of an auction, there are typically cases where data must be queried in realtime e.g. snipe bidding, current bid etc.

The tech stack I used for this site was: [Prisma](https://github.com/prisma/prisma1) w/ [GraphQL-Yoga](https://github.com/dotansimha/graphql-yoga) / [Apollo & GraphQL](https://apollographql.com/) / [Express](https://expressjs.com/) / [NextJS](https://nextjs.org/) / [MongoDB](https://www.mongodb.com/)

The Prisma / GraphQL-Yoga comes with [Subscriptions](https://github.com/apollographql/graphql-subscriptions), and this was the first thing I came across when my client asked for realtime updates. The implementation was quite difficult - took me a long time. Eventually, it was working locally, and in staging. However, when it came to the production environment, for some reason it just didn't work! This unfortunately ended up with a user losing an auction, which ultimately went to court etc.

 I had to fix this mess I had made. I thought it was due to the slow performing queries... After improving query performance, everything was still working in staging and local, so another auction went live, and we figured out that it wasn't due to the slow performing queries. The subscriptions simply didn't work in the production environment.

It must be an environment issue - however I still haven't gotten to the bottom of it. Instead, I've refactored the code and completely removed Subscriptions from the project, replacing them with the [SWR package](https://swr.vercel.app/). The code I have implemented essentially looks something like this:

```
import useSWR from "swr";
import { request } from "graphql-request";
const endpoint = 'http://localhost:4444/graphql'

const fetcher = (query, vars) => request(endpoint, query, vars);

const CURRENT_BID_QUERY = `
     query currentBid {
          // ... data to be returned
     }
`

const vars = {
   // ... query variables
}

const { data, error } = useSWR([CURRENT_BID_QUERY, vars], fetcher, {
    refreshInterval: 1000,
    refreshWhenHidden: true,
    refreshWhenOffline: true
});
```

This polls for data every 1000ms i.e. it performs a query every single second. Now I'm wondering, does this perform a query for every single user, every single second? My concern here is that I am going to run into a query overload type situation, which is the first problem I encountered anyway, so I'd be going backwards here. The only way I can think of testing this out is by running a load test. Does anyone have any advice as to whether or not this is a good idea?

_Imagine the following situation: 100 users all watching the auction listing page. They need to be able to see the current bid in realtime. the useSWR polls for the auction listing every single second, so that it updates for every user._

Is this a bad idea? Will it cause 100 queries per second (due to 100 users)?

**Comment by wwwdepot** [comment]
If server can handle the traffics, it will be fine. Or consider off-load data to [3rd-party providers](https://vercel.com/docs/solutions/realtime#providers) for real-time communications.

**Reply by taylor-lindores-reeves** [reply]
Thank you, I appreciate you reading and providing me with this link. Haven't seen that before!
